Gems/Terrain/Code/Source/Components/TerrainWorldDebuggerComponent.cpp

```diff
@@ -144,178 +158,252 @@ namespace Terrain
+        {
+            AZ::RPI::ViewportContextPtr viewportContext = viewportContextRequests->GetViewportContextById(viewportInfo.m_viewportId);
+            cameraPos = viewportContext->GetCameraTransform().GetTranslation();
+        }
+
+        // Convert our camera position to a wireframe grid sector.  We first convert from world space to sector space by dividing by
+        // sectorSize, so that integer values are sectors, and fractional values are the distance within the sector.  Then we get the
+        // floor, so that we consistently get the next lowest integer - i.e. 2.3 -> 2, and -2.3 -> -3. This gives us consistent behavior
+        // across both positive and negative positions.
+        AZ::Vector2 gridPosition = AZ::Vector2(cameraPos.GetX(), cameraPos.GetY()) / sectorSize;
+        int32_t cameraSectorX = aznumeric_cast<int32_t>(gridPosition.GetFloor().GetX());
+        int32_t cameraSectorY = aznumeric_cast<int32_t>(gridPosition.GetFloor().GetY());
+
+        // Loop through each sector that we *want* to draw, based on camera position.  If the current sector at that index in
+        // m_wireframeSectors doesn't match the world position we want, update its world position and mark it as dirty.
+        // (We loop from -gridSize/2 to gridSize/2 so that the camera is always in the center sector.)
+        for (int32_t sectorY = cameraSectorY - (m_sectorGridSize / 2); sectorY <= cameraSectorY + (m_sectorGridSize / 2); sectorY++)
+        {
+            for (int32_t sectorX = cameraSectorX - (m_sectorGridSize / 2); sectorX <= cameraSectorX + (m_sectorGridSize / 2); sectorX++)
             {
-                if (drawingAabb.Overlaps(sector.m_aabb))
+
+                // Calculate the index in m_wireframeSectors for this sector.  Our indices should rotate through 0 - gridSize, but just
+                // using a single mod will produce a negative result for negative sector indices.  Using abs() will give us incorrect
+                // "backwards" indices for negative numbers, so instead we add the grid size and mod a second time.
+                // Ex:  For a grid size of 5, we want the indices to map like this:
+                //      Index   0   1  2  3  4
+                //      Values -10 -9 -8 -7 -6
+                //             -5  -4 -3 -2 -1
+                //              0   1  2  3  4
+                //              5   6  7  8  9
+                // For -9, (-9 % 5) = -4, then (-4 + 5) % 5 = 1.  If we used abs(), we'd get 4, which is backwards from what we want.
+                int32_t sectorYIndex = ((sectorY % m_sectorGridSize) + m_sectorGridSize) % m_sectorGridSize;
+                int32_t sectorXIndex = ((sectorX % m_sectorGridSize) + m_sectorGridSize) % m_sectorGridSize;
+                int32_t sectorIndex = (sectorYIndex * m_sectorGridSize) + sectorXIndex;
+
+                auto& sector = m_wireframeSectors[sectorIndex];
```

Comment:
nit: per standards I think `WireframeSector&` would be preferable here - it's fairly easy to infer, but since it's not an esoteric type we may as well err on the side of qualification for easier IDE "Go to Definition" queries and the like.

Reply:
Changed!